agent/custom/bbc_action.py

```
@@ -421,20 +458,47 @@ def _setup_and_start_battle(self, team_config: str, run_count: int,
-                mfaalog.info(f"[ExecuteBbcTask] 检测到弹窗: {popup_msgs[0].get('popup_title', '')}")
+            if not state['finished']:
+                popup_msgs = manager.get_messages_by_title('', timeout=2)
+                if popup_msgs:
+                    mfaalog.info(f"[ExecuteBbcTask] 检测到弹窗: {popup_msgs[0].get('popup_title', '')}")
+                    # Handle the queued popup and check if it finished the run
+                    if self._handle_popups(popup_msgs, support_order_mismatch, team_config_error, state, manager):
+                        # Run has finished, stop retrying
+                        return state
         
         if not battle_started:
             error_msg = "启动战斗失败（阵容未设置，已重试3次）"
```

Ruff (0.15.10)

[warning] String contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF001)